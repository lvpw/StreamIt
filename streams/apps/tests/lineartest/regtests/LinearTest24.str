/** 
 * Test out combining larger structures of splitjoins/etc. made
 * up of FIRs.
 **/
void->void pipeline LinearTest24 {
  add FloatSource();
  add pipeline {
    int FIRSIZE = 8;
    add MainBody(10, FIRSIZE);
    add Combiner(10);
  };
  add FloatSink();
}

float->float splitjoin MainBody(int SIZE, int FIRSIZE) {
  split duplicate;
  for (int i=0; i<SIZE; i++) {
    add SimpleAverageFIR(FIRSIZE, 500/(i+1));
  }
  join roundrobin;
}


void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/* simple filter that computes an N point moving average, scaled by a factor A. */
float->float filter SimpleAverageFIR(int N, float A) {
  work push 1 pop 1 peek N {
    float sum = 0;
    for (int i=0; i<N; i++) {
      sum += peek(i);
    }
    push(A * sum / N);
    pop();
  }
}

/* Combines N data elements on the input stream 
 * into a single element on the output stream with addition. */
float->float filter Combiner(int N) {
  work pop N push 1 {
    float sum = 0;
    for (int i=0; i<N; i++) {
      sum += pop();
    }
    push(sum);
  }
}
