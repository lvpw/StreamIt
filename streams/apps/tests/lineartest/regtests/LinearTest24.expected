Combiner_1.matrix
[[1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]]
Combiner_1.vector
[[0.0+0.0i]]
MainBody_1.matrix
[[6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]
 [6.25+0.0i 6.875+0.0i 7.75+0.0i 8.875+0.0i 10.375+0.0i 12.5+0.0i 15.625+0.0i 20.75+0.0i 31.25+0.0i 62.5+0.0i]]
MainBody_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
Pipeline_1.matrix
[[182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]
 [182.75+0.0i]]
Pipeline_1.vector
[[0.0+0.0i]]
SimpleAverageFIR_1.matrix
[[62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]
 [62.5+0.0i]]
SimpleAverageFIR_1.vector
[[0.0+0.0i]]
SimpleAverageFIR_10.matrix
[[6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]
 [6.25+0.0i]]
SimpleAverageFIR_10.vector
[[0.0+0.0i]]
SimpleAverageFIR_2.matrix
[[31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]
 [31.25+0.0i]]
SimpleAverageFIR_2.vector
[[0.0+0.0i]]
SimpleAverageFIR_3.matrix
[[20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]
 [20.75+0.0i]]
SimpleAverageFIR_3.vector
[[0.0+0.0i]]
SimpleAverageFIR_4.matrix
[[15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]
 [15.625+0.0i]]
SimpleAverageFIR_4.vector
[[0.0+0.0i]]
SimpleAverageFIR_5.matrix
[[12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]
 [12.5+0.0i]]
SimpleAverageFIR_5.vector
[[0.0+0.0i]]
SimpleAverageFIR_6.matrix
[[10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]
 [10.375+0.0i]]
SimpleAverageFIR_6.vector
[[0.0+0.0i]]
SimpleAverageFIR_7.matrix
[[8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]
 [8.875+0.0i]]
SimpleAverageFIR_7.vector
[[0.0+0.0i]]
SimpleAverageFIR_8.matrix
[[7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]
 [7.75+0.0i]]
SimpleAverageFIR_8.vector
[[0.0+0.0i]]
SimpleAverageFIR_9.matrix
[[6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]
 [6.875+0.0i]]
SimpleAverageFIR_9.vector
[[0.0+0.0i]]
